b_cfn_sagemaker_endpoint_tests/integration/infrastructure.py

```diff
@@ -0,0 +1,137 @@
+import os
+
+from aws_cdk.aws_iam import Role, ServicePrincipal, PolicyDocument, PolicyStatement, Effect
+from aws_cdk.aws_s3 import Bucket, NotificationKeyFilter, EventType
+from aws_cdk.aws_sagemaker import CfnEndpointConfigProps, CfnModelProps, CfnEndpointProps, CfnModel, CfnEndpointConfig
+from aws_cdk.core import Construct, RemovalPolicy
+from b_aws_testing_framework.tools.cdk_testing.testing_stack import TestingStack
+from b_cfn_s3_large_deployment.deployment_props import DeploymentProps
+from b_cfn_s3_large_deployment.deployment_source import AssetDeploymentSource
+from b_cfn_s3_large_deployment.resource import S3LargeDeploymentResource
+
+from b_cfn_sagemaker_endpoint import SagemakerEndpoint, ModelProps, BucketEvent
+
+
+class Infrastructure(TestingStack):
+    TEST_SAGEMAKER_ENDPOINT_NAME_KEY = 'TestSagemakerEndpointKey'
+
+    def __init__(self, scope: Construct):
+        super().__init__(scope=scope)
+
+        test_bucket = Bucket(
+            scope=self,
+            id=f'{self.global_prefix()}BSagemakerModelsBucket',
+            bucket_name=f'{self.global_prefix()}-b-sagemaker-models'.lower(),
+            auto_delete_objects=True,
+            removal_policy=RemovalPolicy.DESTROY
+        )
+
+        test_role = Role(
```

Comment:
Why are you creating the role? Why can't the model create it on its own?

Reply:
It could but I think it would not be wise/necessary as I'm leaving all the actual configurations of sagemaker components to the end-user, as to not limit his/her use cases.